contents/stable_marriage_problem/code/ruby/stable_marriage.rb

```diff
@@ -0,0 +1,113 @@
+class Generator
+    def initialize(size)
+        @size = size
+    end
+
+    def generate(prefix, r)
+        Array.new(@size){|i|
+            Person.new(
+                i,
+                "#{prefix} #{i}",
+                (0 ... @size).to_a.shuffle!(random: r)
+            )
+        }
+    end
+end
+
+class Person
+    def initialize(id, name, prefs)
+        @id      = id
+        @name    = name
+        @prefs   = prefs
+        @partner = LONELY
+        @choices = 0
+    end
+
+    def lonely
+        @partner == LONELY
+    end
+
+    def propose(partners)
+        if !self.lonely
+            raise '%s is not lonely!' % self.name
+        end
+        choice = @prefs[@choices]
+        partners[choice].onPropose(self)
+        @choices += 1
+    end
+
+    # Represents this Person as an Array of strings
+    def print
+        name = if self.lonely
+            then "Lonely"
+            else @partner.name
+        end
+        [
+            "##{@id} | #{@name}",
+            "Partner: #{name}"
+        ]
+    end
+
+    protected
+    attr_reader :id, :name
+    attr_writer :partner
+
+    # Acts upon a given Proposal
+    def onPropose(partner)
+        if !self.lonely
+            offer = score(partner)
+            current = score(@partner)
+            return unless offer > current 
+            @partner.partner = LONELY
+        end
+        @partner = partner
+        partner.partner = self
+    end
+
+    private
+    LONELY = nil
+
+    # Determines the preference of a given partner
+    def score(partner)
+        return 0 if partner == nil
+        @prefs.size - @prefs.index(partner.id)
+    end
+end
+
+# Deterministic Output, feel free to change seed
+r = Random.new(42)
+
+# Determines Output Columns
+cols = 4
+gen = Generator.new(4)
+men = gen.generate("Man", r)
+women = gen.generate("Woman", r)
+
+# Assume no Name is longer than 20 characters
+spacer = '-' * 20 * cols 
+
+round = 0
+# Solve the Problem
+loop do
```

Comment:
```suggestion
1.step do |round|
```

(and of course remove the definition/modification of `round`)